### PR TITLE
Nd working branch

### PR DESCRIPTION
**Pull request by SaraDawner2000**
Create database for industrial grade photogram

### CODE REVIEW

File: app/views/photos/_form.html.erb
Comment (Cindellai) [inline app/views/photos/_form.html.erb:1]: All of your code looks really good. It's easily readible, consistent spacing, and you don't have any commented out code left in. Great job!

**Review by Cindellai — APPROVED** [review]
good jobbbbb

**Review by Cindellai — APPROVED** [review]
looks great

File: Gemfile [hunk 1]
```diff
@@ -106,3 +106,17 @@ group :test do
   gem "rspec-html-matchers"
   gem "webmock"
 end
+
+group :rubocop do
```
Comment (thierrychau) [inline Gemfile:110]: Oooooohhh Rubocop. Interesting!!

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#  index_likes_on_photo_id  (photo_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (fan_id => users.id)
+#  fk_rails_...  (photo_id => photos.id)
+#
+class Like < ApplicationRecord
+  belongs_to :fan, class_name: "User", counter_cache: true
+  belongs_to :photo, counter_cache: true
```
Comment (thierrychau) [inline app/models/like.rb:23]: Can you add a validation to make sure a user is not able to like the same photo more than once?

File: app/models/photo.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#
+#  index_photos_on_owner_id  (owner_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (owner_id => users.id)
+#
+class Photo < ApplicationRecord
+  validates :caption, :image, presence: true
+
```
Comment (thierrychau) [inline app/models/photo.rb:24]: We want to add a validation on the image too, to make sure there is one

File: app/models/photo.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  belongs_to :owner, class_name: "User", counter_cache: true
+
+  has_many :comments
+
```
Comment (thierrychau) [inline app/models/photo.rb:28]: can you add the association for likes?

File: config.ru [hunk 5]
```diff
@@ -9,7 +9,7 @@ if Rails.env.development?
 end
 
 map '/' do
-	run Rails.application
+	 run Rails.application
```
Comment (thierrychau) [inline config.ru:12]: Was this necessary?

**Review by thierrychau — CHANGES_REQUESTED** [review]
Great job overall, Natalie! I have some minor comments, please see below.

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+         :recoverable, :rememberable, :validatable
+
+  has_many :own_photos, class_name: "Photo", foreign_key: "owner_id"
+
+  has_many :comments, foreign_key: "author_id"
+
+  has_many :likes, foreign_key: "fan_id"
+
+  has_many :sent_follow_requests, class_name: "FollowRequest", foreign_key: "sender_id"
+  has_many :accepted_sent_follow_requests, -> { accepted }, class_name: "FollowRequest", foreign_key: "sender_id"
```
Comment (rezahusain) [inline app/models/user.rb:40]: I'd add a destroy method on this line which makes sure that the sent follow request is deleted when the user who sent the follow request is also deleted. There's more details on this in the project description! If you get stuck then just ask me and i'll help you out :)

**Review by rezahusain — CHANGES_REQUESTED** [review]
Overall I liked that you stuck with what was expected of you and didn't try to add in too much noise. Interesting idea utilizing Rubocop, I'm sure it helped a lot in identifying errors and possible inconsistencies with spellings. I did point out the destroy method issue and that's a feature that you'll likely have to implement soon as it can be a real pain having to deal with as your database gets bigger and users start getting deleted. Fix that and I'll approve of your pull request!

File: app/models/photo.rb (hunk 3 above)
Comment (SaraDawner2000) [reply app/models/photo.rb:24]: It's there, just on the same line

File: config.ru (hunk 5 above)
Comment (SaraDawner2000) [reply config.ru:12]: don't know why that was done, but I didn't do it manually

**Review by rezahusain — APPROVED** [review]
Looks good!

File: app/models/photo.rb (hunk 3 above)
Comment (thierrychau) [reply app/models/photo.rb:24]: Ooooooh, my bad! Great job!

File: config.ru (hunk 5 above)
Comment (thierrychau) [reply config.ru:12]: Make sure to review the files being staged, for this time it's all good though.

**Review by thierrychau — APPROVED** [review]
Good to merge, great job!

**Pull request merged**